testing/test_assertion.py

```diff
@@ -1205,13 +1206,34 @@ def test_truncates_at_8_lines_when_first_8_lines_are_LT_max_chars(self) -> None:
         last_line_before_trunc_msg = result[-self.LINES_IN_TRUNCATION_MSG - 1]
         assert last_line_before_trunc_msg.endswith("...")
 
+    def test_truncates_at_8_lines_when_there_is_one_line_to_remove(self) -> None:
+        """The number of line in the result is 9, the same number as if we truncated."""
+        expl = ["a" for x in range(9)]
+        result = truncate._truncate_explanation(expl, max_lines=8, max_chars=8 * 80)
+        assert result == expl
+        assert "truncated" not in result[-1]
+
+    def test_truncates_edgecase_when_truncation_message_makes_the_result_longer_for_chars(
+        self,
+    ) -> None:
+        expl = ["a" * 10 for x in range(2)]
+        result = truncate._truncate_explanation(expl, max_lines=10, max_chars=10)
+        assert result == ["aaaaaaaaaa", "aaaaaaaaaa"]
+
+    def test_truncates_edgecase_when_truncation_message_makes_the_result_longer_for_lines(
+        self,
+    ) -> None:
+        expl = ["a" * 10 for x in range(2)]
+        result = truncate._truncate_explanation(expl, max_lines=1, max_chars=100)
+        assert result == ["aaaaaaaaaa", "aaaaaaaaaa"]
```

Comment:
Same as before:

```suggestion
        line = "a" * 10
        expl = [line, line]
        result = truncate._truncate_explanation(expl, max_lines=1, max_chars=100)
        assert result == [line, line]
```

Reply:
Do you want me to refactor the existing tests like this in another pull request first ?

Reply:
Did not realize other tests were following the same pattern.

I think it is OK to change the new tests now, and possibly open a new PR refactoring the others. :+1: